### PR TITLE
UI: HDS adoption replace AlertBanner part 2

### PR DESCRIPTION
**Pull request by hellobontempo**
This PR replaces the remaining uses of `<AlertBanner>`. The final PR will address follow-up design items and removing the old AlertBanner component

 Things to lookout for:
- [ ] If no custom title, does the Alert have the correct default (`Info`, `Warning` or `Error`)
- [ ] _Most_ `<Hds::Alert>` components should have `class="has-bottom-margin-s" ` added as a class
- [ ] Are there copy/pasta typos

### CODE REVIEW

File: ui/app/templates/components/token-expire-warning.hbs [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
+    <A.Title>Error</A.Title>
+    <A.Description>
+      Your auth token expired on
+      {{date-format @expirationDate "MMMM do yyyy, h:mm:ss a"}}. You will need to re-authenticate.
+    </A.Description>
+    <A.Link::Standalone @icon="arrow-right" @iconPosition="trailing" @text="Reauthenticate" @route="vault.cluster.logout" />
+  </Hds::Alert>
 {{else}}
-  {{yield}}
+  <section class="section">
```
Comment (hellobontempo) [inline ui/app/templates/components/token-expire-warning.hbs:11]: Moved this here so the banner would span the whole UI width

image

File: ui/app/templates/components/token-expire-warning.hbs [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
+    <div class="container is-widescreen">
+      {{yield}}
+    </div>
+  </section>
```
Comment (hellobontempo) [inline ui/app/templates/components/token-expire-warning.hbs:15]: Moved `<section>` and `<div>` here so the banner spans the whole UI width

image

File: ui/app/templates/components/identity/item-details.hbs [hunk 3]
```diff
@@ -1,17 +1,16 @@
...
+      <A.Description>
+        This
+        {{@model.identityType}}
+        is disabled. All associated tokens cannot be used, but are not revoked.
+      </A.Description>
       {{#if @model.canEdit}}
-        <button onclick={{action "enable" @model}} type="button" class="link" data-test-enable={{true}}>
-          Enable
-        </button>
+        <A.Button data-test-enable-identity @text="Enable" @color="secondary" {{on "click" (action "enable" @model)}} />
```
Comment (hellobontempo) [inline ui/app/templates/components/identity/item-details.hbs:11]: Consulted design and intentionally changed this button to look like a button (not a link)
Screenshot 2023-06-14 at 4 04 17 PM

[secondary](https://helios.hashicorp.design/components/alert#buttons) class is for alert action buttons

File: ui/app/templates/components/keymgmt/distribute.hbs
Comment (hellobontempo) [inline ui/app/templates/components/keymgmt/distribute.hbs:1]: moved up to align with standard form pattern

File: ui/lib/core/addon/templates/components/replication-dashboard.hbs [hunk 4]
```diff
@@ -1,13 +1,20 @@
...
   {{#if this.isReindexing}}
     <div class="reindexing-alert">
-      <AlertBanner
-        @title={{concat "Re-indexing in progress" this.reindexingStage}}
-        @type="info"
-        @progressBar={{this.progressBar}}
-        @message={{sanitized-html this.reindexMessage}}
-        data-test-isReindexing
-      />
+      <Hds::Alert data-test-isReindexing @type="inline" @color="highlight" as |A|>
```
Comment (hellobontempo) [inline ui/lib/core/addon/templates/components/replication-dashboard.hbs:4]: Screenshot 2023-06-14 at 4 52 57 PM

File: ui/lib/core/addon/templates/components/shamir-flow.hbs [hunk 5]
```diff
@@ -138,19 +138,20 @@
...
+            <Hds::Alert @type="inline" @color="highlight" class="has-bottom-margin-s" as |A|>
+              <A.Title>Info</A.Title>
+              <A.Description>
+                Below is the generated OTP. This will be used to encode the generated Operation Token. Make sure to save
+                this, as you will need it later to decode the Operation Token.
+              </A.Description>
+            </Hds::Alert>
           </p>
-          <div class="message is-list has-copy-button" tabindex="-1">
+          <div class="has-background-gray-100 box has-copy-button" tabindex="-1">
```
Comment (hellobontempo) [inline ui/lib/core/addon/templates/components/shamir-flow.hbs:149]: `class="message"` defaults to the old structure's `info` blue background, so this looked a little incohesive beside the new alert banner: 
### before

image

### after: 
updated to generic grey and removed the erroneous div for now. design has a ticket to revisit this 
Screenshot 2023-06-15 at 3 41 00 PM

**Review by zofskeez — APPROVED** [review]
The screenshots in your comments were very helpful thanks for them! Great work 🚀

File: ui/app/templates/components/identity/edit-form.hbs [hunk 6]
```diff
@@ -18,10 +18,10 @@
     <NamespaceReminder @mode={{this.mode}} @noun={{lowercase (humanize this.model.identityType)}} />
     <MessageError @model={{this.model}} />
     {{#if (eq this.mode "merge")}}
-      <AlertBanner
-        @type="warning"
-        @message="Metadata on merged entities is not preserved, you will need to recreate it on the entity you merge to."
-      />
+      <Hds::Alert @type="inline" @color="warning" class="has-bottom-margin-s" as |A|>
```
[the remark this answers is not in the document]
Comment (hellobontempo) [reply ui/app/templates/components/identity/edit-form.hbs:21]: Yep, the original component had a `message` class that included a margin. (All of the other internal styling is handled by HDS, which is so nice!) Hoping to cleanup or delete `message.scss` at the end of this 🤞 
```css
.message {
  background: $blue-010;
  border: 1px solid $blue-100;
  margin-bottom: $spacing-s;
  padding: $spacing-s $spacing-m $spacing-m $spacing-s;
  position: relative;
}
```

**Pull request merged**